From 42c2c795ebc07043ee046dc95d1da5320a02e2ae Mon Sep 17 00:00:00 2001
From: Michel Pollet <buserror@gmail.com>
Date: Wed, 25 Mar 2009 17:38:41 +0000
Subject: [PATCH] VENDOR: armworks logo

Sponsor's kernel logo

Signed-off-by: Michel Pollet <buserror@gmail.com>
---
 drivers/video/logo/Kconfig                   |    5 +
 drivers/video/logo/Makefile                  |    1 +
 drivers/video/logo/logo.c                    |    6 +
 drivers/video/logo/logo_armworks_clut224.ppm |12004 ++++++++++++++++++++++++++
 4 files changed, 12016 insertions(+), 0 deletions(-)
 create mode 100644 drivers/video/logo/logo_armworks_clut224.ppm

diff --git a/drivers/video/logo/Kconfig b/drivers/video/logo/Kconfig
index 39ac49e..34834ed 100644
--- a/drivers/video/logo/Kconfig
+++ b/drivers/video/logo/Kconfig
@@ -82,4 +82,9 @@ config LOGO_M32R_CLUT224
 	depends on M32R
 	default y
 
+config LOGO_ARMWORKS_CLUT224
+	bool "Armworks 224-color logo"
+	depends on MACH_MINI2440
+	default y
+
 endif # LOGO
diff --git a/drivers/video/logo/Makefile b/drivers/video/logo/Makefile
index b91251d..4d0423f 100644
--- a/drivers/video/logo/Makefile
+++ b/drivers/video/logo/Makefile
@@ -15,6 +15,7 @@ obj-$(CONFIG_LOGO_SUPERH_MONO)		+= logo_superh_mono.o
 obj-$(CONFIG_LOGO_SUPERH_VGA16)		+= logo_superh_vga16.o
 obj-$(CONFIG_LOGO_SUPERH_CLUT224)	+= logo_superh_clut224.o
 obj-$(CONFIG_LOGO_M32R_CLUT224)		+= logo_m32r_clut224.o
+obj-$(CONFIG_LOGO_ARMWORKS_CLUT224)	+= logo_armworks_clut224.o
 
 obj-$(CONFIG_SPU_BASE)			+= logo_spe_clut224.o
 
diff --git a/drivers/video/logo/logo.c b/drivers/video/logo/logo.c
index 2e85a2b..451817e 100644
--- a/drivers/video/logo/logo.c
+++ b/drivers/video/logo/logo.c
@@ -35,6 +35,7 @@ extern const struct linux_logo logo_superh_mono;
 extern const struct linux_logo logo_superh_vga16;
 extern const struct linux_logo logo_superh_clut224;
 extern const struct linux_logo logo_m32r_clut224;
+extern const struct linux_logo logo_armworks_clut224;
 
 static int nologo;
 module_param(nologo, bool, 0);
@@ -115,6 +116,11 @@ const struct linux_logo * __init_refok fb_find_logo(int depth)
 		/* M32R Linux logo */
 		logo = &logo_m32r_clut224;
 #endif
+#ifdef CONFIG_LOGO_ARMWORKS_CLUT224
+		/* ARMWorks Linux logo -- lower priority */
+		if (!logo)
+			logo = &logo_armworks_clut224;
+#endif
 	}
 	return logo;
 }
diff --git a/drivers/video/logo/logo_armworks_clut224.ppm b/drivers/video/logo/logo_armworks_clut224.ppm
new file mode 100644
index 0000000..a632167
--- /dev/null
+++ b/drivers/video/logo/logo_armworks_clut224.ppm
@@ -0,0 +1,12004 @@
+P3
+# CREATOR: GIMP PNM Filter Version 1.1
+160 25
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+237
+229
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+222
+222
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+239
+236
+254
+249
+242
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+219
+190
+253
+70
+115
+253
+70
+115
+253
+248
+253
+255
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+107
+98
+252
+0
+46
+254
+105
+154
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+245
+233
+253
+26
+69
+251
+51
+102
+253
+252
+250
+254
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+194
+181
+253
+0
+46
+254
+31
+58
+250
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+252
+250
+254
+74
+79
+250
+96
+142
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+252
+250
+254
+255
+246
+253
+107
+108
+252
+51
+102
+253
+54
+95
+252
+39
+86
+253
+54
+95
+252
+94
+132
+253
+196
+207
+254
+252
+250
+254
+254
+255
+252
+248
+253
+255
+252
+250
+254
+146
+155
+253
+113
+137
+251
+252
+250
+254
+254
+255
+252
+254
+255
+252
+252
+250
+254
+255
+246
+253
+51
+102
+253
+206
+211
+254
+252
+250
+254
+93
+105
+254
+92
+119
+252
+146
+172
+254
+252
+250
+254
+248
+253
+255
+254
+255
+252
+252
+250
+254
+120
+126
+252
+168
+190
+254
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+105
+113
+251
+92
+119
+252
+121
+164
+254
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+255
+240
+254
+1
+52
+251
+5
+39
+251
+214
+226
+253
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+201
+165
+255
+31
+58
+250
+0
+58
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+74
+79
+250
+0
+63
+251
+121
+164
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+234
+219
+253
+120
+126
+252
+51
+102
+253
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+172
+152
+253
+31
+58
+250
+0
+46
+254
+213
+233
+255
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+181
+166
+251
+31
+58
+250
+0
+74
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+250
+226
+255
+33
+46
+250
+1
+52
+251
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+33
+46
+250
+0
+63
+251
+0
+63
+251
+0
+63
+251
+31
+58
+250
+12
+65
+254
+1
+52
+251
+94
+132
+253
+254
+255
+252
+254
+255
+252
+252
+250
+254
+31
+58
+250
+0
+46
+254
+241
+254
+255
+254
+255
+252
+254
+255
+252
+248
+253
+255
+194
+166
+255
+0
+58
+254
+117
+152
+253
+254
+255
+252
+107
+108
+252
+0
+63
+251
+25
+80
+252
+255
+252
+251
+255
+252
+251
+254
+255
+252
+252
+231
+254
+33
+46
+250
+39
+86
+253
+254
+255
+252
+254
+255
+252
+254
+255
+252
+241
+214
+253
+1
+52
+251
+31
+58
+250
+176
+178
+253
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+231
+254
+1
+52
+251
+1
+52
+251
+223
+230
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+201
+170
+252
+0
+63
+251
+0
+63
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+74
+79
+250
+0
+63
+251
+129
+161
+253
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+120
+126
+252
+31
+58
+250
+70
+115
+253
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+240
+254
+39
+86
+253
+70
+115
+253
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+178
+163
+255
+0
+63
+251
+0
+74
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+132
+122
+252
+26
+69
+251
+0
+63
+251
+155
+174
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+246
+253
+31
+58
+250
+31
+58
+250
+113
+137
+251
+197
+177
+253
+146
+155
+253
+52
+67
+251
+26
+69
+251
+33
+46
+250
+214
+226
+253
+254
+255
+252
+255
+246
+253
+1
+52
+251
+0
+63
+251
+155
+174
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+84
+94
+254
+0
+63
+251
+51
+102
+253
+254
+255
+252
+215
+182
+254
+31
+58
+250
+1
+52
+251
+227
+239
+254
+248
+253
+255
+254
+255
+252
+186
+158
+253
+0
+63
+251
+1
+52
+251
+227
+239
+254
+252
+250
+254
+254
+255
+252
+172
+152
+253
+0
+63
+251
+1
+52
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+231
+254
+0
+58
+254
+0
+58
+254
+222
+222
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+201
+170
+252
+31
+58
+250
+0
+63
+251
+254
+255
+252
+249
+242
+255
+212
+203
+255
+249
+242
+255
+252
+250
+254
+171
+171
+253
+159
+159
+253
+188
+197
+251
+252
+250
+254
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+255
+246
+253
+168
+164
+253
+159
+159
+253
+206
+211
+254
+65
+74
+252
+0
+63
+251
+129
+161
+253
+254
+255
+252
+211
+207
+253
+211
+207
+253
+255
+246
+253
+248
+253
+255
+254
+255
+252
+252
+250
+254
+220
+206
+252
+211
+207
+253
+248
+253
+255
+254
+255
+252
+245
+233
+253
+168
+164
+253
+159
+159
+253
+191
+191
+253
+255
+252
+251
+222
+222
+254
+105
+113
+251
+31
+58
+250
+54
+95
+252
+237
+229
+255
+220
+206
+252
+248
+253
+255
+239
+236
+254
+211
+207
+253
+222
+222
+254
+254
+255
+252
+168
+164
+253
+159
+159
+253
+191
+191
+253
+222
+222
+254
+237
+229
+255
+234
+219
+253
+254
+255
+252
+254
+255
+252
+255
+246
+253
+194
+181
+253
+159
+159
+253
+168
+164
+253
+206
+211
+254
+252
+250
+254
+254
+255
+252
+254
+255
+252
+178
+163
+255
+31
+58
+250
+0
+74
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+248
+253
+255
+255
+252
+251
+0
+46
+254
+0
+63
+251
+19
+66
+255
+26
+90
+253
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+31
+58
+250
+31
+58
+250
+168
+190
+254
+255
+252
+251
+248
+253
+255
+255
+235
+252
+0
+58
+254
+0
+63
+251
+176
+178
+253
+254
+255
+252
+219
+190
+253
+31
+58
+250
+12
+65
+254
+51
+102
+253
+252
+250
+254
+254
+255
+252
+255
+252
+251
+1
+52
+251
+26
+69
+251
+0
+58
+254
+248
+253
+255
+252
+255
+251
+1
+52
+251
+31
+58
+250
+121
+164
+254
+254
+255
+252
+254
+255
+252
+65
+74
+252
+26
+69
+251
+31
+58
+250
+146
+172
+254
+248
+253
+255
+255
+252
+251
+74
+79
+250
+31
+58
+250
+55
+116
+253
+254
+255
+252
+254
+255
+252
+254
+255
+252
+203
+186
+253
+159
+159
+253
+171
+171
+253
+239
+236
+254
+252
+250
+254
+248
+253
+255
+255
+252
+251
+211
+207
+253
+211
+207
+253
+254
+255
+252
+206
+193
+253
+159
+159
+253
+171
+171
+253
+245
+233
+253
+31
+58
+250
+0
+58
+254
+223
+230
+252
+252
+250
+254
+254
+255
+252
+220
+206
+252
+223
+212
+252
+188
+197
+251
+254
+255
+252
+254
+255
+252
+171
+171
+253
+159
+159
+253
+176
+178
+253
+249
+242
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+201
+165
+255
+0
+63
+251
+31
+58
+250
+248
+253
+255
+197
+177
+253
+1
+52
+251
+25
+80
+252
+1
+52
+251
+1
+52
+251
+0
+58
+254
+1
+52
+251
+39
+86
+253
+254
+255
+252
+252
+250
+254
+248
+253
+255
+197
+177
+253
+0
+46
+254
+0
+63
+251
+1
+52
+251
+0
+58
+254
+52
+67
+251
+31
+58
+250
+121
+164
+254
+252
+231
+254
+0
+46
+254
+0
+46
+254
+214
+226
+253
+255
+252
+251
+254
+255
+252
+240
+208
+253
+0
+46
+254
+5
+39
+251
+230
+244
+251
+172
+152
+253
+1
+52
+251
+31
+58
+250
+31
+58
+250
+0
+58
+254
+0
+74
+252
+105
+113
+251
+0
+63
+251
+26
+69
+251
+0
+63
+251
+1
+52
+251
+0
+46
+254
+214
+226
+253
+178
+163
+255
+1
+52
+251
+0
+63
+251
+0
+63
+251
+0
+58
+254
+31
+58
+250
+1
+52
+251
+1
+52
+251
+31
+58
+250
+0
+46
+254
+232
+249
+255
+254
+255
+252
+33
+46
+250
+0
+58
+254
+31
+58
+250
+0
+63
+251
+31
+58
+250
+0
+74
+252
+252
+250
+254
+254
+255
+252
+181
+166
+251
+0
+63
+251
+0
+74
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+201
+170
+252
+0
+63
+251
+26
+69
+251
+52
+67
+251
+1
+52
+251
+222
+222
+254
+252
+250
+254
+254
+255
+252
+255
+246
+253
+1
+52
+251
+0
+58
+254
+170
+195
+252
+255
+252
+251
+255
+252
+251
+255
+240
+254
+1
+52
+251
+31
+58
+250
+176
+178
+253
+254
+255
+252
+186
+158
+253
+0
+63
+251
+26
+69
+251
+1
+52
+251
+254
+255
+252
+254
+255
+252
+215
+182
+254
+31
+58
+250
+52
+67
+251
+1
+52
+251
+223
+230
+252
+252
+250
+254
+107
+98
+252
+0
+63
+251
+84
+94
+254
+254
+255
+252
+242
+226
+254
+1
+52
+251
+0
+63
+251
+26
+69
+251
+26
+90
+253
+255
+252
+251
+255
+240
+254
+1
+52
+251
+0
+63
+251
+151
+183
+252
+255
+252
+251
+220
+206
+252
+1
+52
+251
+0
+58
+254
+31
+58
+250
+0
+63
+251
+0
+46
+254
+94
+132
+253
+254
+255
+252
+249
+242
+255
+5
+39
+251
+1
+52
+251
+26
+69
+251
+1
+52
+251
+0
+63
+251
+33
+46
+250
+222
+222
+254
+0
+58
+254
+0
+58
+254
+214
+226
+253
+254
+255
+252
+139
+132
+253
+1
+52
+251
+5
+39
+251
+168
+190
+254
+220
+206
+252
+33
+46
+250
+31
+58
+250
+31
+58
+250
+31
+58
+250
+5
+39
+251
+232
+237
+253
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+201
+170
+252
+0
+63
+251
+31
+58
+250
+252
+250
+254
+197
+177
+253
+0
+63
+251
+12
+65
+254
+52
+67
+251
+159
+159
+253
+105
+113
+251
+0
+58
+254
+31
+58
+250
+82
+136
+253
+254
+255
+252
+210
+185
+254
+1
+52
+251
+0
+63
+251
+26
+69
+251
+159
+159
+253
+67
+107
+251
+0
+63
+251
+0
+63
+251
+121
+164
+254
+255
+235
+252
+1
+52
+251
+1
+52
+251
+223
+230
+252
+254
+255
+252
+254
+255
+252
+234
+219
+253
+31
+58
+250
+1
+52
+251
+248
+253
+255
+0
+46
+254
+26
+69
+251
+51
+102
+253
+181
+166
+251
+105
+113
+251
+96
+142
+252
+176
+178
+253
+64
+85
+253
+26
+69
+251
+26
+69
+251
+137
+146
+254
+92
+119
+252
+230
+244
+251
+181
+166
+251
+0
+63
+251
+52
+67
+251
+1
+52
+251
+146
+155
+253
+78
+97
+251
+145
+140
+253
+54
+95
+252
+0
+63
+251
+0
+58
+254
+241
+254
+255
+254
+255
+252
+145
+140
+253
+92
+119
+252
+159
+159
+253
+120
+126
+252
+0
+58
+254
+31
+58
+250
+82
+136
+253
+254
+255
+252
+181
+166
+251
+31
+58
+250
+0
+74
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+83
+87
+252
+0
+63
+251
+113
+167
+252
+105
+113
+251
+0
+63
+251
+96
+142
+252
+248
+253
+255
+254
+255
+252
+255
+246
+253
+31
+58
+250
+31
+58
+250
+146
+172
+254
+252
+231
+254
+220
+206
+252
+84
+94
+254
+26
+69
+251
+0
+46
+254
+230
+244
+251
+254
+255
+252
+93
+105
+254
+0
+63
+251
+26
+69
+251
+0
+58
+254
+173
+186
+252
+255
+252
+251
+105
+113
+251
+0
+63
+251
+0
+63
+251
+31
+58
+250
+191
+191
+253
+254
+255
+252
+197
+177
+253
+0
+63
+251
+1
+52
+251
+248
+253
+249
+186
+158
+253
+0
+63
+251
+31
+58
+250
+26
+69
+251
+1
+52
+251
+230
+244
+251
+181
+166
+251
+31
+58
+250
+0
+46
+254
+241
+254
+255
+252
+231
+254
+5
+39
+251
+26
+69
+251
+26
+69
+251
+171
+171
+253
+105
+113
+251
+0
+58
+254
+0
+58
+254
+146
+172
+254
+255
+252
+251
+1
+52
+251
+26
+69
+251
+31
+58
+250
+51
+102
+253
+145
+140
+253
+77
+122
+252
+255
+240
+254
+1
+52
+251
+1
+52
+251
+232
+237
+253
+186
+158
+253
+0
+58
+254
+1
+52
+251
+146
+172
+254
+252
+250
+254
+64
+85
+253
+52
+67
+251
+25
+80
+252
+171
+171
+253
+145
+140
+253
+26
+69
+251
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+201
+165
+255
+0
+63
+251
+0
+63
+251
+254
+255
+252
+197
+177
+253
+31
+58
+250
+31
+58
+250
+252
+255
+251
+248
+253
+255
+254
+255
+252
+132
+122
+252
+0
+63
+251
+0
+74
+252
+254
+255
+252
+52
+67
+251
+26
+69
+251
+39
+86
+253
+252
+255
+251
+254
+255
+252
+255
+252
+251
+74
+79
+250
+31
+58
+250
+129
+161
+253
+255
+235
+252
+0
+58
+254
+0
+58
+254
+222
+222
+254
+254
+255
+252
+255
+252
+251
+241
+214
+253
+0
+58
+254
+1
+52
+251
+252
+250
+254
+1
+52
+251
+0
+63
+251
+155
+174
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+113
+137
+251
+31
+58
+250
+55
+116
+253
+248
+253
+255
+254
+255
+252
+252
+250
+254
+186
+158
+253
+0
+63
+251
+0
+58
+254
+206
+211
+254
+254
+255
+252
+252
+255
+251
+255
+252
+251
+229
+192
+251
+0
+63
+251
+1
+52
+251
+241
+254
+255
+255
+252
+251
+248
+253
+255
+254
+255
+252
+252
+250
+254
+252
+250
+254
+186
+158
+253
+0
+63
+251
+25
+80
+252
+252
+250
+254
+181
+166
+251
+0
+63
+251
+26
+69
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+250
+226
+255
+0
+58
+254
+1
+52
+251
+214
+226
+253
+194
+166
+255
+0
+63
+251
+0
+58
+254
+255
+252
+251
+254
+255
+252
+255
+252
+251
+33
+46
+250
+26
+69
+251
+31
+58
+250
+0
+58
+254
+31
+58
+250
+26
+69
+251
+1
+52
+251
+144
+180
+254
+248
+253
+255
+254
+255
+252
+52
+67
+251
+52
+67
+251
+31
+58
+250
+26
+69
+251
+55
+116
+253
+252
+250
+254
+1
+52
+251
+26
+69
+251
+0
+63
+251
+26
+69
+251
+96
+142
+252
+248
+253
+255
+252
+250
+254
+0
+46
+254
+0
+63
+251
+170
+195
+252
+83
+87
+252
+0
+63
+251
+144
+180
+254
+65
+74
+252
+31
+58
+250
+144
+180
+254
+107
+98
+252
+0
+63
+251
+51
+102
+253
+254
+255
+252
+132
+122
+252
+0
+63
+251
+0
+63
+251
+254
+255
+252
+248
+253
+255
+255
+252
+251
+105
+113
+251
+52
+67
+251
+0
+63
+251
+254
+255
+252
+1
+52
+251
+52
+67
+251
+55
+116
+253
+255
+252
+251
+254
+255
+252
+255
+252
+251
+252
+231
+254
+0
+58
+254
+31
+58
+250
+206
+193
+253
+1
+52
+251
+0
+63
+251
+114
+145
+251
+254
+255
+252
+254
+255
+252
+52
+67
+251
+0
+63
+251
+77
+122
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+254
+255
+252
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+194
+166
+255
+31
+58
+250
+0
+63
+251
+252
+255
+251
+197
+177
+253
+0
+63
+251
+0
+63
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+201
+170
+252
+31
+58
+250
+0
+74
+252
+254
+255
+252
+1
+52
+251
+31
+58
+250
+188
+197
+251
+248
+253
+255
+255
+252
+251
+255
+252
+251
+74
+79
+250
+0
+63
+251
+121
+164
+254
+255
+240
+254
+0
+58
+254
+0
+58
+254
+223
+230
+252
+248
+253
+255
+255
+252
+251
+241
+214
+253
+0
+58
+254
+1
+52
+251
+248
+253
+249
+83
+87
+252
+0
+63
+251
+0
+58
+254
+54
+95
+252
+191
+191
+253
+254
+255
+252
+255
+252
+251
+120
+126
+252
+0
+63
+251
+67
+107
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+181
+166
+251
+0
+63
+251
+0
+58
+254
+248
+253
+255
+255
+252
+251
+252
+255
+251
+252
+250
+254
+215
+182
+254
+0
+63
+251
+1
+52
+251
+241
+254
+255
+255
+252
+251
+203
+186
+253
+50
+77
+252
+0
+46
+254
+33
+46
+250
+0
+58
+254
+52
+67
+251
+26
+69
+251
+252
+250
+254
+181
+166
+251
+0
+63
+251
+0
+74
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+145
+140
+253
+0
+63
+251
+0
+63
+251
+255
+252
+251
+255
+246
+253
+1
+52
+251
+31
+58
+250
+151
+183
+252
+255
+252
+251
+255
+252
+251
+1
+52
+251
+0
+63
+251
+26
+69
+251
+0
+63
+251
+0
+63
+251
+0
+58
+254
+176
+178
+253
+254
+255
+252
+252
+250
+254
+255
+235
+252
+0
+58
+254
+31
+58
+250
+206
+193
+253
+1
+52
+251
+0
+58
+254
+240
+208
+253
+0
+58
+254
+26
+69
+251
+186
+158
+253
+0
+63
+251
+70
+115
+253
+255
+252
+251
+254
+255
+252
+84
+94
+254
+26
+69
+251
+25
+80
+252
+0
+63
+251
+1
+52
+251
+227
+239
+254
+173
+144
+251
+0
+63
+251
+64
+85
+253
+26
+69
+251
+0
+63
+251
+155
+174
+251
+254
+255
+252
+50
+77
+252
+0
+63
+251
+82
+136
+253
+255
+252
+251
+252
+250
+254
+255
+252
+251
+229
+192
+251
+0
+58
+254
+1
+52
+251
+252
+255
+251
+1
+52
+251
+31
+58
+250
+206
+193
+253
+248
+253
+255
+255
+252
+251
+254
+255
+252
+252
+231
+254
+1
+52
+251
+0
+63
+251
+0
+58
+254
+26
+69
+251
+1
+52
+251
+248
+253
+255
+254
+255
+252
+255
+252
+251
+163
+139
+252
+0
+63
+251
+0
+63
+251
+0
+74
+252
+146
+172
+254
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+248
+253
+255
+201
+165
+255
+0
+63
+251
+31
+58
+250
+255
+252
+251
+197
+177
+253
+0
+58
+254
+0
+63
+251
+252
+250
+254
+254
+255
+252
+252
+250
+254
+201
+170
+252
+0
+63
+251
+50
+77
+252
+255
+246
+253
+1
+52
+251
+0
+58
+254
+196
+207
+254
+255
+252
+251
+254
+255
+252
+254
+255
+252
+74
+79
+250
+0
+63
+251
+121
+164
+254
+252
+231
+254
+1
+52
+251
+0
+58
+254
+214
+226
+253
+254
+255
+252
+254
+255
+252
+241
+214
+253
+31
+58
+250
+0
+58
+254
+219
+245
+255
+252
+255
+251
+107
+108
+252
+1
+52
+251
+0
+63
+251
+0
+58
+254
+54
+95
+252
+254
+255
+252
+120
+126
+252
+31
+58
+250
+67
+107
+251
+248
+253
+255
+254
+255
+252
+252
+255
+251
+178
+163
+255
+0
+63
+251
+25
+80
+252
+254
+255
+252
+248
+253
+255
+252
+250
+254
+254
+255
+252
+223
+186
+252
+0
+58
+254
+0
+58
+254
+248
+253
+249
+197
+177
+253
+33
+46
+250
+52
+67
+251
+0
+63
+251
+94
+132
+253
+78
+97
+251
+0
+63
+251
+0
+74
+252
+254
+255
+252
+181
+166
+251
+0
+63
+251
+26
+69
+251
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+31
+58
+250
+52
+67
+251
+77
+122
+252
+220
+206
+252
+223
+212
+252
+65
+74
+252
+0
+63
+251
+26
+90
+253
+248
+253
+255
+255
+246
+253
+1
+52
+251
+31
+58
+250
+176
+205
+255
+254
+255
+252
+0
+63
+251
+26
+69
+251
+0
+74
+252
+254
+255
+252
+254
+255
+252
+230
+202
+255
+0
+58
+254
+0
+58
+254
+254
+255
+252
+52
+67
+251
+52
+67
+251
+50
+77
+252
+0
+63
+251
+96
+142
+252
+219
+190
+253
+0
+58
+254
+1
+52
+251
+252
+250
+254
+248
+253
+249
+181
+166
+251
+0
+63
+251
+26
+69
+251
+52
+67
+251
+39
+86
+253
+252
+255
+251
+242
+226
+254
+31
+58
+250
+0
+63
+251
+26
+69
+251
+1
+52
+251
+238
+247
+255
+248
+253
+255
+65
+74
+252
+31
+58
+250
+121
+164
+254
+255
+252
+251
+254
+255
+252
+254
+255
+252
+240
+208
+253
+0
+58
+254
+1
+52
+251
+239
+236
+254
+1
+52
+251
+0
+63
+251
+222
+222
+254
+254
+255
+252
+254
+255
+252
+248
+253
+255
+252
+231
+254
+0
+58
+254
+52
+67
+251
+0
+63
+251
+0
+63
+251
+26
+69
+251
+94
+132
+253
+248
+253
+255
+248
+253
+255
+254
+255
+252
+155
+151
+252
+0
+46
+254
+31
+58
+250
+31
+58
+250
+19
+66
+255
+252
+250
+254
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+194
+166
+255
+0
+63
+251
+31
+58
+250
+254
+255
+252
+197
+177
+253
+31
+58
+250
+0
+63
+251
+254
+255
+252
+254
+255
+252
+252
+250
+254
+201
+170
+252
+0
+63
+251
+0
+74
+252
+255
+252
+251
+1
+52
+251
+31
+58
+250
+188
+197
+251
+254
+255
+252
+248
+253
+255
+254
+255
+252
+74
+79
+250
+31
+58
+250
+121
+164
+254
+245
+233
+253
+1
+52
+251
+1
+52
+251
+223
+230
+252
+252
+250
+254
+254
+255
+252
+250
+226
+255
+1
+52
+251
+1
+52
+251
+227
+239
+254
+254
+255
+252
+254
+255
+252
+255
+240
+254
+93
+105
+254
+0
+63
+251
+0
+58
+254
+168
+190
+254
+145
+140
+253
+0
+63
+251
+67
+107
+251
+248
+253
+255
+252
+250
+254
+254
+255
+252
+181
+166
+251
+31
+58
+250
+0
+74
+252
+255
+252
+251
+254
+255
+252
+248
+253
+255
+248
+253
+255
+223
+186
+252
+31
+58
+250
+1
+52
+251
+248
+253
+249
+64
+85
+253
+0
+63
+251
+55
+116
+253
+255
+252
+251
+248
+253
+255
+220
+206
+252
+1
+52
+251
+0
+74
+252
+252
+250
+254
+181
+166
+251
+0
+63
+251
+0
+74
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+215
+182
+254
+0
+58
+254
+52
+67
+251
+0
+63
+251
+0
+58
+254
+0
+58
+254
+0
+63
+251
+26
+69
+251
+1
+52
+251
+213
+233
+255
+255
+252
+251
+31
+58
+250
+31
+58
+250
+168
+190
+254
+255
+252
+251
+230
+202
+255
+1
+52
+251
+31
+58
+250
+114
+145
+251
+254
+255
+252
+173
+144
+251
+0
+63
+251
+50
+77
+252
+248
+253
+249
+172
+152
+253
+0
+63
+251
+26
+69
+251
+0
+58
+254
+195
+216
+245
+252
+231
+254
+0
+58
+254
+1
+52
+251
+227
+239
+254
+255
+252
+251
+255
+240
+254
+1
+52
+251
+26
+69
+251
+31
+58
+250
+121
+164
+254
+243
+245
+242
+254
+255
+252
+50
+77
+252
+52
+67
+251
+52
+67
+251
+26
+90
+253
+255
+252
+251
+255
+252
+251
+64
+85
+253
+0
+63
+251
+94
+132
+253
+254
+255
+252
+248
+253
+255
+254
+255
+252
+210
+185
+254
+0
+63
+251
+1
+52
+251
+248
+253
+249
+0
+46
+254
+0
+58
+254
+196
+207
+254
+254
+255
+252
+255
+252
+251
+254
+255
+252
+252
+231
+254
+0
+58
+254
+52
+67
+251
+77
+122
+252
+132
+122
+252
+0
+63
+251
+33
+46
+250
+222
+222
+254
+254
+255
+252
+255
+252
+251
+252
+255
+251
+254
+255
+252
+145
+140
+253
+0
+63
+251
+0
+58
+254
+77
+122
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+255
+251
+201
+170
+252
+0
+63
+251
+0
+63
+251
+254
+255
+252
+197
+177
+253
+0
+63
+251
+0
+63
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+201
+165
+255
+31
+58
+250
+50
+77
+252
+252
+250
+254
+52
+67
+251
+0
+63
+251
+25
+80
+252
+254
+255
+252
+255
+252
+251
+243
+245
+242
+74
+79
+250
+0
+63
+251
+129
+161
+253
+255
+246
+253
+1
+52
+251
+31
+58
+250
+168
+190
+254
+254
+255
+252
+252
+255
+251
+163
+139
+252
+0
+63
+251
+1
+52
+251
+230
+244
+251
+237
+229
+255
+248
+253
+255
+254
+255
+252
+255
+252
+251
+31
+58
+250
+0
+63
+251
+121
+164
+254
+113
+137
+251
+0
+63
+251
+26
+90
+253
+254
+255
+252
+254
+255
+252
+248
+253
+249
+178
+163
+255
+0
+63
+251
+26
+69
+251
+255
+252
+251
+254
+255
+252
+255
+252
+251
+255
+252
+251
+210
+185
+254
+0
+63
+251
+0
+58
+254
+254
+255
+252
+26
+69
+251
+31
+58
+250
+144
+180
+254
+254
+255
+252
+252
+250
+254
+163
+139
+252
+0
+63
+251
+0
+58
+254
+254
+255
+252
+186
+158
+253
+31
+58
+250
+50
+77
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+83
+87
+252
+0
+63
+251
+26
+69
+251
+146
+155
+253
+137
+146
+254
+137
+146
+254
+155
+151
+252
+0
+58
+254
+0
+63
+251
+113
+137
+251
+255
+252
+251
+33
+46
+250
+0
+63
+251
+170
+195
+252
+252
+255
+251
+255
+252
+251
+120
+113
+253
+26
+69
+251
+33
+46
+250
+227
+239
+254
+105
+113
+251
+31
+58
+250
+77
+122
+252
+252
+250
+254
+250
+226
+255
+31
+58
+250
+0
+63
+251
+26
+69
+251
+255
+252
+251
+252
+255
+251
+1
+52
+251
+31
+58
+250
+176
+178
+253
+255
+252
+251
+248
+253
+249
+64
+85
+253
+0
+63
+251
+1
+52
+251
+223
+230
+252
+255
+252
+251
+252
+250
+254
+186
+158
+253
+0
+63
+251
+0
+58
+254
+129
+161
+253
+255
+252
+251
+252
+250
+254
+139
+132
+253
+0
+63
+251
+0
+63
+251
+252
+250
+254
+255
+252
+251
+248
+253
+255
+74
+79
+250
+0
+63
+251
+0
+74
+252
+255
+252
+251
+0
+58
+254
+0
+58
+254
+206
+211
+254
+252
+250
+254
+254
+255
+252
+243
+245
+242
+252
+231
+254
+0
+58
+254
+1
+52
+251
+232
+233
+243
+248
+253
+249
+31
+58
+250
+26
+69
+251
+0
+74
+252
+248
+253
+255
+249
+242
+255
+232
+237
+253
+254
+255
+252
+254
+255
+252
+132
+122
+252
+0
+63
+251
+51
+102
+253
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+252
+250
+254
+252
+250
+254
+201
+165
+255
+0
+63
+251
+31
+58
+250
+255
+252
+251
+197
+177
+253
+31
+58
+250
+0
+63
+251
+254
+255
+252
+254
+255
+252
+248
+253
+255
+194
+166
+255
+31
+58
+250
+0
+74
+252
+252
+250
+254
+172
+152
+253
+0
+58
+254
+0
+63
+251
+0
+58
+254
+105
+113
+251
+50
+77
+252
+0
+63
+251
+52
+67
+251
+129
+161
+253
+252
+255
+251
+52
+67
+251
+0
+63
+251
+0
+58
+254
+67
+107
+251
+65
+74
+252
+31
+58
+250
+52
+67
+251
+1
+52
+251
+248
+253
+249
+107
+98
+252
+0
+63
+251
+113
+137
+251
+107
+108
+252
+0
+63
+251
+0
+58
+254
+191
+191
+253
+186
+158
+253
+31
+58
+250
+0
+63
+251
+26
+90
+253
+64
+85
+253
+117
+152
+253
+197
+177
+253
+0
+63
+251
+0
+74
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+255
+252
+251
+215
+182
+254
+0
+58
+254
+0
+58
+254
+252
+255
+251
+78
+97
+251
+52
+67
+251
+1
+52
+251
+113
+137
+251
+93
+105
+254
+0
+58
+254
+52
+67
+251
+1
+52
+251
+248
+253
+249
+181
+166
+251
+0
+63
+251
+1
+52
+251
+223
+230
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+252
+231
+254
+31
+58
+250
+0
+63
+251
+117
+152
+253
+255
+252
+251
+255
+252
+251
+243
+245
+242
+252
+255
+251
+105
+113
+251
+0
+63
+251
+0
+63
+251
+254
+255
+252
+1
+52
+251
+0
+58
+254
+168
+190
+254
+248
+253
+255
+255
+252
+251
+248
+253
+255
+0
+58
+254
+0
+63
+251
+51
+102
+253
+50
+77
+252
+0
+63
+251
+146
+172
+254
+252
+255
+251
+255
+252
+251
+83
+87
+252
+0
+63
+251
+129
+161
+253
+252
+250
+254
+254
+255
+252
+83
+87
+252
+0
+63
+251
+105
+154
+254
+252
+250
+254
+254
+255
+252
+186
+158
+253
+0
+63
+251
+39
+86
+253
+254
+255
+252
+252
+255
+251
+252
+255
+251
+255
+235
+252
+0
+58
+254
+1
+52
+251
+227
+239
+254
+254
+255
+252
+254
+255
+252
+255
+246
+253
+0
+46
+254
+0
+63
+251
+0
+63
+251
+113
+137
+251
+50
+77
+252
+52
+67
+251
+1
+52
+251
+170
+195
+252
+255
+252
+251
+1
+52
+251
+31
+58
+250
+206
+211
+254
+254
+255
+252
+248
+253
+255
+252
+255
+251
+252
+231
+254
+0
+58
+254
+1
+52
+251
+237
+229
+255
+252
+255
+251
+230
+202
+255
+1
+52
+251
+31
+58
+250
+105
+154
+254
+194
+166
+255
+1
+52
+251
+77
+122
+252
+113
+137
+251
+31
+58
+250
+0
+63
+251
+82
+136
+253
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+186
+158
+253
+33
+46
+250
+0
+46
+254
+254
+255
+252
+201
+170
+252
+33
+46
+250
+0
+46
+254
+254
+255
+252
+254
+255
+252
+255
+252
+251
+201
+165
+255
+1
+52
+251
+31
+58
+250
+254
+255
+252
+252
+255
+251
+139
+132
+253
+1
+52
+251
+0
+63
+251
+1
+52
+251
+0
+58
+254
+31
+58
+250
+0
+46
+254
+105
+154
+254
+255
+252
+251
+223
+212
+252
+1
+52
+251
+0
+58
+254
+0
+58
+254
+1
+52
+251
+78
+97
+251
+1
+52
+251
+5
+39
+251
+241
+254
+255
+26
+69
+251
+1
+52
+251
+1
+52
+251
+0
+58
+254
+1
+52
+251
+94
+132
+253
+252
+255
+251
+254
+255
+252
+50
+77
+252
+1
+52
+251
+31
+58
+250
+1
+52
+251
+129
+161
+253
+168
+164
+253
+33
+46
+250
+31
+58
+250
+252
+250
+254
+254
+255
+252
+252
+250
+254
+254
+255
+252
+201
+170
+252
+1
+52
+251
+5
+39
+251
+241
+254
+255
+255
+240
+254
+0
+63
+251
+0
+58
+254
+0
+58
+254
+0
+46
+254
+77
+122
+252
+26
+69
+251
+0
+58
+254
+77
+122
+252
+241
+214
+253
+0
+46
+254
+0
+58
+254
+39
+86
+253
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+249
+139
+132
+253
+33
+46
+250
+33
+46
+250
+227
+239
+254
+255
+252
+251
+255
+252
+251
+248
+253
+255
+255
+252
+251
+229
+192
+251
+33
+46
+250
+5
+39
+251
+151
+183
+252
+33
+46
+250
+33
+46
+250
+151
+183
+252
+255
+252
+251
+252
+250
+254
+255
+252
+251
+210
+185
+254
+33
+46
+250
+1
+52
+251
+1
+52
+251
+5
+39
+251
+196
+207
+254
+252
+250
+254
+248
+253
+255
+201
+170
+252
+32
+30
+249
+227
+239
+254
+254
+255
+252
+254
+255
+252
+120
+126
+252
+0
+46
+254
+26
+69
+251
+254
+255
+252
+252
+250
+254
+255
+235
+252
+5
+39
+251
+105
+154
+254
+254
+255
+252
+252
+250
+254
+248
+253
+255
+248
+253
+255
+74
+79
+250
+26
+69
+251
+252
+250
+254
+252
+250
+254
+248
+253
+255
+255
+252
+251
+250
+226
+255
+52
+67
+251
+1
+52
+251
+0
+58
+254
+1
+52
+251
+0
+58
+254
+155
+174
+251
+252
+250
+254
+255
+240
+254
+5
+39
+251
+0
+46
+254
+196
+207
+254
+254
+255
+252
+254
+255
+252
+248
+253
+255
+252
+231
+254
+5
+39
+251
+0
+46
+254
+214
+226
+253
+248
+253
+255
+254
+255
+252
+120
+113
+253
+1
+52
+251
+33
+46
+250
+25
+80
+252
+1
+52
+251
+33
+46
+250
+1
+52
+251
+1
+52
+251
+54
+95
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+245
+233
+253
+245
+233
+253
+248
+253
+255
+248
+253
+255
+245
+233
+253
+245
+233
+253
+254
+255
+252
+252
+250
+254
+248
+253
+255
+252
+250
+254
+237
+229
+255
+245
+233
+253
+255
+252
+251
+252
+255
+251
+254
+255
+252
+249
+242
+255
+211
+207
+253
+211
+207
+253
+243
+245
+242
+255
+246
+253
+245
+233
+253
+249
+242
+255
+255
+252
+251
+254
+255
+252
+254
+255
+252
+220
+206
+252
+211
+207
+253
+232
+233
+243
+252
+255
+251
+237
+229
+255
+245
+233
+253
+248
+253
+255
+255
+252
+251
+237
+229
+255
+220
+206
+252
+211
+207
+253
+234
+239
+242
+252
+250
+254
+248
+253
+255
+254
+255
+252
+254
+255
+252
+220
+206
+252
+220
+206
+252
+223
+212
+252
+249
+242
+255
+252
+250
+254
+237
+229
+255
+239
+236
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+242
+226
+254
+237
+229
+255
+252
+255
+251
+254
+255
+252
+254
+255
+252
+223
+212
+252
+211
+207
+253
+245
+233
+253
+248
+253
+255
+255
+246
+253
+211
+207
+253
+240
+235
+233
+254
+255
+252
+242
+226
+254
+211
+207
+253
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+252
+250
+254
+239
+236
+254
+242
+226
+254
+237
+229
+255
+252
+255
+251
+248
+253
+255
+254
+255
+252
+252
+250
+254
+248
+253
+255
+252
+255
+251
+245
+233
+253
+242
+226
+254
+249
+242
+255
+245
+233
+253
+242
+226
+254
+252
+250
+254
+254
+255
+252
+255
+252
+251
+252
+255
+251
+248
+253
+249
+252
+231
+254
+245
+233
+253
+245
+233
+253
+237
+229
+255
+254
+255
+252
+255
+252
+251
+248
+253
+255
+248
+253
+249
+211
+207
+253
+252
+250
+254
+248
+253
+255
+254
+255
+252
+249
+242
+255
+237
+229
+255
+239
+236
+254
+254
+255
+252
+255
+252
+251
+252
+255
+251
+220
+206
+252
+252
+250
+254
+254
+255
+252
+255
+252
+251
+254
+255
+252
+252
+255
+251
+237
+229
+255
+237
+229
+255
+254
+255
+252
+255
+252
+251
+252
+250
+254
+248
+253
+255
+252
+255
+251
+252
+255
+251
+234
+219
+253
+211
+207
+253
+223
+212
+252
+243
+245
+242
+254
+255
+252
+252
+255
+251
+252
+255
+251
+245
+233
+253
+237
+229
+255
+248
+253
+255
+252
+250
+254
+255
+252
+251
+248
+253
+255
+248
+253
+249
+242
+226
+254
+237
+229
+255
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+237
+229
+255
+245
+233
+253
+255
+252
+251
+249
+242
+255
+220
+206
+252
+211
+207
+253
+239
+236
+254
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+255
+252
+251
+252
+250
+254
+252
+255
+251
+252
+255
+251
+252
+250
+254
+252
+250
+254
+248
+253
+255
+252
+250
+254
+254
+255
+252
+255
+252
+251
+248
+253
+255
+255
+252
+251
+248
+253
+255
+252
+250
+254
+252
+250
+254
+255
+252
+251
+255
+252
+251
+254
+255
+252
+255
+252
+251
+254
+255
+252
+252
+255
+251
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+248
+253
+255
+254
+255
+252
+255
+252
+251
+248
+253
+255
+255
+252
+251
+248
+253
+255
+255
+252
+251
+248
+253
+255
+255
+252
+251
+252
+255
+251
+255
+252
+251
+252
+250
+254
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+255
+252
+251
+248
+253
+255
+248
+253
+255
+252
+255
+251
+248
+253
+255
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+252
+250
+254
+252
+255
+251
+248
+253
+249
+255
+252
+251
+255
+252
+251
+248
+253
+255
+255
+252
+251
+252
+250
+254
+248
+253
+255
+252
+250
+254
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+252
+255
+251
+254
+255
+252
+252
+250
+254
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+248
+253
+255
+254
+255
+252
+252
+255
+251
+248
+253
+255
+254
+255
+252
+248
+253
+255
+248
+253
+255
+248
+253
+255
+252
+250
+254
+252
+250
+254
+248
+253
+255
+248
+253
+255
+248
+253
+255
+255
+252
+251
+248
+253
+255
+255
+252
+251
+254
+255
+252
+252
+250
+254
+252
+250
+254
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+255
+252
+251
+255
+252
+251
+248
+253
+255
+248
+253
+255
+255
+252
+251
+248
+253
+255
+254
+255
+252
+254
+255
+252
+248
+253
+255
+248
+253
+255
+252
+255
+251
+252
+255
+251
+248
+253
+255
+254
+255
+252
+255
+252
+251
+254
+255
+252
+255
+252
+251
+252
+250
+254
+255
+252
+251
+248
+253
+249
+252
+255
+251
+248
+253
+255
+255
+252
+251
+252
+250
+254
+252
+250
+254
+248
+253
+249
+248
+253
+255
+255
+252
+251
+255
+252
+251
+254
+255
+252
+255
+252
+251
+255
+252
+251
+255
+252
+251
+252
+255
+251
+255
+252
+251
+254
+255
+252
+254
+255
+252
+252
+250
+254
+248
+253
+255
+254
+255
+252
+248
+253
+255
+254
+255
+252
+252
+255
+251
+252
+255
+251
+255
+252
+251
+252
+250
+254
+252
+255
+251
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+255
+252
+251
+254
+255
+252
+248
+253
+255
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+255
+252
+251
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+254
+255
+252
+252
+250
+254
+254
+255
+252
+252
+250
+254
+252
+250
+254
+255
+252
+251
+254
+255
+252
+252
+250
+254
+252
+250
+254
+254
+255
+252
+255
+252
+251
+255
+252
+251
+254
+255
+252
+255
+252
+251
+254
+255
+252
+255
+252
+251
+254
+255
+252
+255
+252
+251
+255
+252
+251
+248
+253
+255
+254
+255
+252
+254
+255
+252
+252
+250
+254
+252
+250
+254
+248
+253
+255
+230
+227
+225
+248
+253
+255
+234
+239
+242
+201
+189
+179
+238
+247
+255
+255
+252
+251
+254
+255
+252
+207
+202
+200
+252
+250
+254
+252
+250
+254
+254
+255
+252
+230
+227
+225
+219
+225
+231
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+252
+250
+254
+250
+242
+234
+182
+177
+176
+248
+253
+255
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+252
+250
+254
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+252
+250
+254
+255
+252
+251
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+243
+245
+242
+234
+239
+242
+255
+252
+251
+255
+252
+251
+254
+255
+252
+252
+250
+254
+254
+255
+252
+248
+253
+255
+255
+252
+251
+255
+252
+251
+254
+255
+252
+234
+226
+218
+211
+217
+219
+254
+255
+252
+255
+252
+251
+255
+252
+251
+254
+255
+252
+255
+252
+251
+243
+245
+242
+254
+255
+252
+254
+255
+252
+240
+235
+233
+230
+227
+225
+234
+239
+242
+248
+253
+255
+254
+255
+252
+254
+255
+252
+250
+242
+234
+205
+213
+222
+255
+252
+251
+252
+250
+254
+254
+255
+252
+248
+253
+255
+254
+255
+252
+223
+214
+206
+234
+239
+242
+252
+250
+254
+197
+193
+191
+254
+255
+252
+252
+250
+254
+248
+253
+255
+254
+255
+252
+210
+205
+197
+248
+253
+255
+255
+252
+251
+232
+233
+243
+243
+245
+242
+252
+250
+254
+252
+255
+251
+254
+255
+252
+255
+252
+251
+252
+250
+254
+255
+252
+251
+250
+242
+234
+205
+213
+222
+254
+255
+252
+254
+255
+252
+255
+252
+251
+248
+253
+255
+248
+253
+255
+252
+250
+254
+252
+250
+254
+252
+250
+254
+254
+255
+252
+255
+252
+251
+255
+252
+251
+255
+252
+251
+252
+250
+254
+252
+250
+254
+252
+250
+254
+248
+253
+255
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+248
+253
+255
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+254
+255
+252
+248
+253
+255
+252
+250
+254
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+101
+81
+79
+254
+255
+252
+93
+90
+106
+223
+214
+206
+248
+253
+255
+182
+177
+176
+234
+239
+242
+133
+131
+123
+191
+195
+199
+248
+253
+255
+248
+253
+255
+114
+98
+88
+133
+131
+123
+133
+153
+169
+223
+214
+206
+205
+213
+222
+250
+242
+234
+211
+217
+219
+127
+119
+113
+168
+191
+209
+223
+214
+206
+191
+206
+219
+250
+242
+234
+217
+211
+210
+219
+225
+231
+191
+195
+199
+234
+226
+218
+205
+213
+222
+234
+226
+218
+191
+195
+199
+254
+255
+252
+217
+211
+210
+191
+195
+199
+254
+255
+252
+197
+193
+191
+205
+213
+222
+223
+214
+206
+197
+217
+223
+254
+255
+252
+218
+199
+188
+133
+153
+169
+254
+255
+252
+223
+214
+206
+191
+206
+219
+234
+226
+218
+248
+253
+255
+219
+225
+231
+230
+227
+225
+248
+253
+255
+233
+220
+201
+71
+78
+85
+135
+121
+116
+230
+244
+251
+182
+177
+176
+238
+247
+255
+234
+226
+218
+200
+207
+211
+99
+93
+91
+191
+206
+219
+254
+255
+252
+131
+111
+91
+110
+114
+118
+211
+217
+219
+217
+211
+210
+200
+207
+211
+207
+202
+200
+253
+243
+227
+109
+128
+145
+191
+195
+199
+254
+255
+252
+197
+193
+191
+238
+247
+255
+233
+220
+201
+127
+119
+113
+214
+226
+253
+210
+205
+197
+83
+94
+106
+254
+255
+252
+191
+195
+199
+248
+253
+255
+223
+214
+206
+116
+95
+90
+219
+245
+255
+241
+216
+191
+103
+132
+168
+233
+200
+174
+127
+173
+205
+218
+199
+188
+205
+213
+222
+250
+242
+234
+211
+217
+219
+243
+245
+242
+165
+156
+151
+144
+145
+154
+219
+225
+231
+234
+226
+218
+255
+252
+251
+211
+217
+219
+201
+189
+179
+219
+225
+231
+250
+242
+234
+211
+217
+219
+243
+245
+242
+191
+195
+199
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+128
+106
+98
+136
+138
+135
+116
+128
+152
+175
+164
+146
+140
+125
+114
+191
+206
+219
+114
+126
+132
+152
+151
+143
+190
+169
+150
+144
+172
+189
+252
+250
+254
+143
+119
+115
+136
+138
+135
+163
+169
+165
+99
+101
+107
+110
+114
+118
+175
+164
+146
+149
+165
+177
+135
+121
+116
+201
+189
+179
+123
+140
+148
+159
+140
+125
+158
+152
+139
+133
+153
+169
+168
+130
+120
+182
+180
+166
+116
+128
+152
+122
+122
+119
+210
+205
+197
+114
+111
+108
+166
+173
+180
+136
+125
+125
+117
+106
+105
+174
+167
+156
+149
+165
+177
+201
+189
+179
+99
+101
+107
+99
+101
+107
+241
+254
+255
+218
+199
+188
+144
+172
+189
+253
+243
+227
+104
+107
+127
+165
+156
+151
+152
+151
+143
+144
+145
+154
+83
+94
+106
+114
+111
+108
+252
+250
+254
+189
+172
+142
+171
+197
+224
+248
+253
+255
+157
+123
+106
+197
+217
+223
+114
+111
+108
+117
+106
+105
+166
+173
+180
+114
+111
+108
+219
+225
+231
+248
+253
+255
+140
+125
+114
+142
+155
+162
+240
+235
+233
+119
+121
+124
+114
+111
+108
+175
+164
+146
+149
+165
+177
+142
+155
+162
+158
+152
+139
+135
+121
+116
+123
+140
+148
+133
+131
+123
+96
+109
+136
+152
+151
+143
+174
+167
+156
+135
+154
+178
+122
+124
+136
+146
+125
+108
+152
+151
+143
+122
+124
+136
+130
+146
+146
+146
+125
+108
+232
+249
+255
+253
+243
+227
+86
+88
+91
+117
+106
+105
+171
+197
+224
+174
+167
+156
+86
+88
+91
+182
+180
+166
+135
+154
+178
+117
+106
+105
+210
+205
+197
+152
+156
+171
+122
+122
+119
+116
+95
+90
+149
+143
+139
+167
+182
+195
+149
+143
+139
+96
+107
+119
+165
+156
+151
+167
+182
+195
+101
+81
+79
+133
+131
+123
+176
+200
+211
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+114
+98
+88
+254
+255
+252
+104
+107
+127
+175
+164
+146
+210
+205
+197
+119
+121
+124
+114
+126
+132
+136
+138
+135
+233
+220
+201
+128
+158
+189
+248
+253
+255
+99
+93
+91
+248
+253
+255
+248
+253
+255
+117
+106
+105
+142
+155
+162
+174
+157
+146
+204
+224
+243
+146
+125
+108
+227
+239
+254
+110
+114
+118
+110
+114
+118
+190
+169
+150
+204
+224
+243
+168
+130
+120
+230
+228
+206
+128
+158
+189
+158
+152
+139
+140
+125
+114
+133
+131
+123
+167
+182
+195
+144
+145
+154
+162
+139
+130
+217
+211
+210
+122
+124
+136
+142
+155
+162
+135
+121
+116
+143
+146
+145
+248
+253
+255
+223
+214
+206
+86
+88
+91
+123
+140
+148
+136
+125
+125
+99
+101
+107
+248
+253
+249
+83
+74
+80
+216
+201
+179
+103
+132
+168
+254
+255
+252
+250
+242
+234
+99
+93
+91
+119
+121
+124
+201
+189
+179
+110
+114
+118
+142
+155
+162
+165
+156
+151
+96
+107
+119
+158
+152
+139
+135
+154
+178
+254
+255
+252
+146
+125
+108
+119
+121
+124
+211
+217
+219
+142
+155
+162
+136
+125
+125
+230
+228
+206
+144
+172
+189
+110
+114
+118
+96
+107
+119
+216
+201
+179
+122
+124
+136
+182
+177
+176
+122
+122
+119
+99
+93
+91
+217
+211
+210
+99
+101
+107
+96
+107
+119
+201
+189
+179
+130
+146
+146
+166
+173
+180
+127
+119
+113
+99
+93
+91
+219
+245
+255
+250
+233
+208
+103
+132
+168
+233
+200
+174
+144
+172
+189
+99
+101
+107
+110
+114
+118
+190
+169
+150
+195
+216
+245
+175
+152
+137
+114
+111
+108
+144
+172
+189
+174
+139
+122
+176
+200
+211
+83
+74
+80
+230
+228
+206
+96
+107
+119
+110
+114
+118
+165
+156
+151
+230
+244
+251
+159
+140
+125
+144
+145
+154
+200
+207
+211
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+254
+255
+252
+255
+252
+251
+255
+252
+251
+254
+255
+252
+234
+226
+218
+136
+125
+125
+176
+200
+211
+252
+250
+254
+248
+253
+255
+254
+255
+252
+252
+250
+254
+255
+252
+251
+255
+252
+251
+248
+253
+255
+255
+252
+251
+248
+253
+255
+248
+253
+255
+255
+252
+251
+254
+255
+252
+255
+252
+251
+248
+253
+255
+248
+253
+255
+254
+255
+252
+248
+253
+255
+248
+253
+255
+252
+250
+254
+248
+253
+255
+254
+255
+252
+252
+250
+254
+254
+255
+252
+252
+250
+254
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+252
+250
+254
+254
+255
+252
+255
+252
+251
+255
+252
+251
+255
+252
+251
+248
+253
+255
+254
+255
+252
+248
+253
+255
+248
+253
+255
+248
+253
+255
+252
+255
+251
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+248
+253
+255
+254
+255
+252
+255
+252
+251
+248
+253
+255
+252
+250
+254
+252
+250
+254
+254
+255
+252
+254
+255
+252
+255
+252
+251
+248
+253
+255
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+255
+252
+251
+254
+255
+252
+255
+252
+251
+252
+250
+254
+248
+253
+255
+248
+253
+255
+255
+252
+251
+248
+253
+255
+248
+253
+255
+255
+252
+251
+252
+250
+254
+255
+252
+251
+255
+252
+251
+248
+253
+255
+254
+255
+252
+248
+253
+255
+255
+252
+251
+255
+252
+251
+252
+250
+254
+254
+255
+252
+254
+255
+252
+255
+252
+251
+255
+252
+251
+248
+253
+255
+254
+255
+252
+248
+253
+255
+248
+253
+255
+254
+255
+252
+254
+255
+252
+252
+250
+254
+255
+252
+251
+252
+250
+254
+252
+250
+254
+248
+253
+255
+255
+252
+251
+248
+253
+255
+254
+255
+252
+254
+255
+252
+248
+253
+255
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+248
+253
+255
+248
+253
+255
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+255
+252
+251
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+248
+253
+255
+254
+255
+252
+252
+255
+251
+254
+255
+252
+248
+253
+255
+254
+255
+252
+254
+255
+252
+255
+252
+251
+248
+253
+255
+254
+255
+252
+254
+255
+252
+255
+252
+251
+255
+252
+251
+254
+255
+252
+255
+252
+251
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+254
+255
+252
+255
+252
+251
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+255
+252
+251
+255
+252
+251
+255
+252
+251
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+255
+252
+251
+252
+250
+254
+255
+252
+251
+254
+255
+252
+252
+255
+251
+254
+255
+252
+255
+252
+251
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+248
+253
+255
+254
+255
+252
+255
+252
+251
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+248
+253
+255
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+255
+252
+251
+252
+250
+254
+254
+255
+252
+254
+255
+252
+248
+253
+255
+254
+255
+252
+248
+253
+255
+254
+255
+252
+255
+252
+251
+252
+250
+254
+254
+255
+252
+254
+255
+252
+248
+253
+255
+255
+252
+251
+254
+255
+252
+252
+250
+254
+252
+250
+254
+255
+252
+251
+254
+255
+252
+254
+255
+252
+248
+253
+255
+255
+252
+251
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+248
+253
+255
+255
+252
+251
+254
+255
+252
+252
+250
+254
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+255
+252
+251
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+255
+252
+251
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+252
+250
+254
+248
+253
+255
+252
+250
+254
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+252
+250
+254
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
+254
+255
+252
-- 
1.5.6.3

